lib/datadog/core/environment/process.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+require_relative 'ext'
+
+module Datadog
+  module Core
+    module Environment
+      # Retrieves process level information
+      # TODO: Add memoization
+      module Process
+        module_function
+
+        def entrypoint_workdir
+          File.basename(Dir.pwd)
+        end
+
+        def entrypoint_type
+          Core::Environment::Ext::PROCESS_TYPE
+        end
+
+        def entrypoint_name
+          File.basename($0.to_s)
```

Comment:
Is there a reason why $0 needs to be stringified?  It looks like [it's already a string](https://github.com/ruby/rbs/blob/031f319726752e74499fc7036c9593a3b5d87ed0/core/global_variables.rbs#L77-L78).